### PR TITLE
ceph: osd various lvm enhancement

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

See the commits.

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

File: pkg/daemon/ceph/osd/volume.go [hunk 1]
```diff
@@ -153,20 +154,14 @@ func updateLVMConfig(context *clusterd.Context, onPVC bool) error {
 	// When running on PVC
 	if onPVC {
 		output = bytes.Replace(output, []byte(`scan = [ "/dev" ]`), []byte(`scan = [ "/dev", "/mnt" ]`), 1)
-		// Only filter blocks in /mnt, when running on PVC we copy the PVC claim path to /mnt
-		// And reject everything else
-		// We have 2 different regex depending on the version of LVM present in the container...
-		// Since https://github.com/lvmteam/lvm2/commit/08396b4bce45fb8311979250623f04ec0ddb628c#diff-13c602a6258e57ce666a240e67c44f38
-		// the content changed, so depending which version is installled one of the two replace will work
-		output = bytes.Replace(output, []byte(`# filter = [ "a|.*/|" ]`), []byte(`filter = [ "a|^/mnt/.*|", "r|.*|" ]`), 1)
-		output = bytes.Replace(output, []byte(`# filter = [ "a|.*|" ]`), []byte(`filter = [ "a|^/mnt/.*|", "r|.*|" ]`), 1)
```
Comment (travisn) [inline pkg/daemon/ceph/osd/volume.go]: Wait, we don't need this different filter depending on the version of lvm?

Comment (leseb) [reply pkg/daemon/ceph/osd/volume.go]: No. I should have gone for that in my previous patch. Actually, we should never have done that from the beginning...

Comment (travisn) [reply pkg/daemon/ceph/osd/volume.go]: I see, we are just overriding the value and no need to look at the original value.

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 2]
```diff
@@ -869,5 +869,6 @@ func cephVolumeEnvVar() []v1.EnvVar {
 	return []v1.EnvVar{
 		{Name: "CEPH_VOLUME_DEBUG", Value: "1"},
 		{Name: "CEPH_VOLUME_SKIP_RESTORECON", Value: "1"},
+		{Name: "DM_DISABLE_UDEV", Value: "1"},
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go]: Can you add a comment here explaining the env var? I see it in the commit message, but would be good to have it next to the code as well.

File: pkg/daemon/ceph/osd/volume.go (hunk 1 above)
Comment (BlaineEXE) [reply pkg/daemon/ceph/osd/volume.go]: :clap:

File: pkg/operator/ceph/cluster/osd/spec.go (hunk 2 above)
Comment (BlaineEXE) [reply pkg/operator/ceph/cluster/osd/spec.go]: Some clarifying questions:
1. What is the benefit of doing it without udev versus with udev?
2. Am I understanding correctly that all ceph-volume based provisioning and OSDs will get this value?
3. Does this change also mean we no longer need to mount `/run/udev` (IIRC) in the container(s)?

Comment (leseb) [reply pkg/operator/ceph/cluster/osd/spec.go]: > Some clarifying questions:
> 
> 1. What is the benefit of doing it without udev versus with udev?

I guess the benefit is to bypass the natural difficulties we have to properly work with udev (exposing various sockets etc). I'm also hoping to remove potential races as well as speeding up LVM by only looking at /dev

> 2. Am I understanding correctly that all ceph-volume based provisioning and OSDs will get this value?

Correct, the filter is needed for both prepare and activate.

> 3. Does this change also mean we no longer need to mount `/run/udev` (IIRC) in the container(s)?

That's a good question, I haven't tried removing it, but I wouldn't mind removing it.
I guess I'm just afraid to break other scenarios (such as encryption).

Does that help?

**Pull request merged**